### ISSUE TITLE
Random crash on saving the profile

### ISSUE THREAD

**Issue by madeye** [shadowsocks/shadowsocks-android#2522]
Please read contributing guidelines. Thanks.

**Describe the bug**
App crashed on saving the profile

**To Reproduce**
Open a profile and click the saving button on the top

**Expected behavior**
Save and exit the current fragment

**Screenshots**
None

**Smartphone (please complete the following information):**
 - Android/Chrome OS version: Android 10
 - Device: Oneplus 8
 - Version: 5.1.x
 - Last version that did not exhibit the issue: 5.0.6

**Additional context**
```
05-10 08:46:36.965 22397 22397 E AndroidRuntime: Process: com.github.shadowsocks, PID: 22397                                                
05-10 08:46:36.965 22397 22397 E AndroidRuntime: java.lang.IllegalStateException: Check failed.                                             
05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at com.github.shadowsocks.database.Profile.deserialize(Profile.kt:365)              
05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at com.github.shadowsocks.ProfileConfigFragment.saveAndExit(ProfileConfigFragment.kt
:148)                                                                                                                                       
05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at com.github.shadowsocks.ProfileConfigFragment.onOptionsItemSelected(ProfileConfigF
ragment.kt:246)                                                                                                                             
05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at com.github.shadowsocks.ProfileConfigActivity.onOptionsItemSelected(ProfileConfigActivity.kt:75)                                                                                                                              
05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at android.app.Activity.onMenuItemSelected(Activity.java:4160)                      05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:3
83)                                                                                                                                         05-10 08:46:36.965 22397 22397 E AndroidRuntime:        at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.jav
a:219)                                                                     
```

**Comment by Mygod** [comment]
Unfortunately I have not reproduced this a single time...

**Comment by madeye** [comment]
Right, I cannot reproduce this every time. Need to figure out the exact steps to reproduce this crash.

**Comment by madeye** [comment]
With the latest update of Oneplus 8, this issue is gone.

**Comment by Mygod** [comment]
Huh amazing?

**Comment by Mygod** [comment]
Might be fixed via c852fc2.
